src/Workspaces/Core/MSBuild/MSBuild/BuildHostProcessManager.cs

```diff
@@ -197,6 +205,23 @@ private void AppendBuildHostCommandLineArgumentsConfigureProcess(ProcessStartInf
         processStartInfo.RedirectStandardError = true;
     }
 
+    private static void AddArgument(ProcessStartInfo processStartInfo, string argument)
+    {
+        // On .NET Core 2.1 and higher we can just use ArgumentList to do the right thing; downlevel we need to manually
+        // construct the string
+#if NET
+        processStartInfo.ArgumentList.Add(argument);
+#else
+        if (processStartInfo.Arguments.Length > 0)
+            processStartInfo.Arguments += ' ';
+
+        if (argument.Contains(' '))
+            processStartInfo.Arguments += '"' + argument + '"';
```

Comment:
maybe cleaner to always quote?

Reply:
This is also used to add regular switches to dotnet and friends. It'd look quite odd if those were escaped (which would make people think there were bugs) and potentially also might trigger bugs if those don't expect it.